recipes-bsp/rpi-eeprom/rpi-eeprom_git.bb

```diff
@@ -0,0 +1,66 @@
+SUMMARY = "Installation scripts and binaries for the Raspberry Pi 4 EEPROM"
+DESCRIPTION = "This repository contains the rpi4 bootloader and scripts \
+for updating it in the spi eeprom"
+LICENSE = "BSD-3-Clause & Broadcom-RPi"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=f546ed4f47e9d4c1fe954ecc9d3ef4f3"
+
+SRC_URI = " \
+    git://github.com/raspberrypi/rpi-eeprom.git;protocol=https;branch=master \
+"
+
+SRCREV = "6e79e995bbc75c5fdd5305bd7fe029758cfade2f"
+PV = "v2022.12.07-138a1"
+
+S = "${WORKDIR}/git"
+
+RDEPENDS:${PN} += " \
+    coreutils \
+    python3 \
+    python3-pycryptodomex \
+    openssl \
+    xxd \
+    pciutils \
+    userland \
+"
+
+inherit python3native
+
+do_install() {
+    install -d ${D}${bindir}
+
+    # install executables
+    install -m 0755 ${S}/tools/vl805 ${D}${bindir}
+    install -m 0755 ${S}/rpi-eeprom-update ${D}${bindir}
+    install -m 0755 ${S}/rpi-eeprom-config ${D}${bindir}
+    install -m 0755 ${S}/rpi-eeprom-digest ${D}${bindir}
+
+    # copy firmware files
+    install -d ${D}${base_libdir}/firmware/raspberrypi/bootloader/critical
+    install -d ${D}${base_libdir}/firmware/raspberrypi/bootloader/stable
+    install -d ${D}${base_libdir}/firmware/raspberrypi/bootloader/beta
+
+    install -m 644 ${S}/firmware/critical/* ${D}${base_libdir}/firmware/raspberrypi/bootloader/critical
+    install -m 644 ${S}/firmware/stable/* ${D}${base_libdir}/firmware/raspberrypi/bootloader/stable
+    install -m 644 ${S}/firmware/beta/* ${D}${base_libdir}/firmware/raspberrypi/bootloader/beta
+
+    ln -s critical ${D}${base_libdir}/firmware/raspberrypi/bootloader/default
+    ln -s stable ${D}${base_libdir}/firmware/raspberrypi/bootloader/latest
+
+    # copy default config
+    install -d ${D}${sysconfdir}/default
+    install -D ${S}/rpi-eeprom-update-default ${D}${sysconfdir}/default/rpi-eeprom-update
+}
+
+FILES:${PN} += "${sbindir}/vl805"
+FILES:${PN} += "${base_libdir}/firmware/raspberrypi/bootloader/*"
+FILES:${PN} += "${sysconfdir}/*"
+
+INHIBIT_PACKAGE_STRIP = "1"
+INHIBIT_PACKAGE_DEBUG_SPLIT = "1"
+
+# vl805 tool sources are not available (yet), as it comes as a precompiled
+# binary only. It has ARM architecture whereas target machine is Aarch64. We
+# need to disable arch check for it otherwise it cannot packed.
+QAPATHTEST[arch] = ""
```

Comment:
Have you tested this binary to work on rpi4 64 bit mode

Reply:
Yes, it's a statically linked executable and it executes on aarch64 (assuming the kernel is configured to support 32-bit userspace applications).
```
% file tools/vl805
tools/vl805: ELF 32-bit LSB executable, ARM, EABI5 version 1 (SYSV), statically linked, for GNU/Linux 3.2.0, BuildID[sha1]=1f534f3dce4bee014889300c7bb329ccda12b13c, stripped
```
```
root@raspberrypi4-64:~# uname -m    
aarch64
root@raspberrypi4-64:~# /usr/bin/vl805 -h
Usage: vl805 [<options>]
  where <options> are taken from:
    -h            Show this help message
    -r <file>     Read EEPROM
    -w <file>     Write EEPROM
    -v <file>     Verify EEPROM
    -o <offset>   Specify an offset to read from/write to (for testing).
                  N.B. Erases to the end of the EEPROM!
    -d            Verbose debugging
```

Reply:
Statically linked ok makes sense